apps/analysis_framework/serializers.py

```diff
@@ -126,15 +130,82 @@ def validate_project(self, project):
 
     def create(self, validated_data):
         project = validated_data.pop('project', None)
+        private = validated_data.get('is_private', False)
+
+        # Check if user has access to private project feature
+        user = self.context['request'].user
+        private_access = user.profile.get_accessible_features().filter(
+            key=Feature.PRIVATE_PROJECT
+        ).exists()
+
+        if private and not private_access:
+            raise exceptions.PermissionDenied({
+                'message': 'You can not create private frameworks.'
+            })
+
         af = super().create(validated_data)
 
         if project:
             project = Project.objects.get(id=project)
             project.analysis_framework = af
-            project.modified_by = self.context['request'].user
+            project.modified_by = user
             project.save()
 
+        owner_role = af.get_or_create_owner_role()
+        af.add_member(self.context['request'].user, owner_role)
         return af
 
+    def update(self, instance, validated_data):
+        if 'is_private' not in validated_data:
+            return super().update(instance, validated_data)
+
+        if instance.is_private != validated_data['is_private']:
+            raise exceptions.PermissionDenied('Cannot change privacy of project')
```

Comment:
Privace of `framework` instead of `project`?

Reply:
Fixed.